src/org/opensolaris/opengrok/history/HistoryGuru.java

```diff
@@ -341,69 +343,95 @@ public void addRepositories(File[] files, Collection<RepositoryInfo> repos,
      */
     private void addRepositories(File[] files, Collection<RepositoryInfo> repos,
             IgnoredNames ignoredNames, boolean recursiveSearch, int depth) {
+
+        // This check for ignoring a repository only works for embedded
```

Comment:
I think the previous algorithm is actually wrong. If the traversal descends into the directory that happens to be a repository, it already added the repository to the list, so the check for the ".opengrok_skip_history" file inside will do nothing. See pull request #1727. Maybe you can change the fix accordingly.